app/helpers/sessions_helper.rb

```diff
@@ -38,7 +39,30 @@ def clear_cart
     end
     
     def log_out
+        user = User.find_by(id: session[:user_id])
+        user.update_attribute(:cart, hto_s(session[:cart]))
+        clear_cart
         session.delete(:user_id)
         @current_user = nil
     end
+    
+    private
+        
+    #converts hashs to string    
+    def hto_s(hash)
+        shash = ""
+        hash.each do |id, quantity|
```

Comment:
@modsoussi Does `hash.to_s` not work? What if the hash is nested? Would you need to recurse?
```ruby
shash += id + " " + hto_s(quantity) + ","
```